apps/server/src/shared/infra/identity-management/keycloak-configuration/console/keycloak-configuration.console.ts

```diff
@@ -112,12 +118,65 @@ export class KeycloakConsole {
 		);
 	}
 
-	private async repeatCommand<T>(
-		commandName: string,
-		command: () => Promise<T>,
-		count: number,
-		delay: number
-	): Promise<T> {
+	/**
+	 * For migration purpose. Moves all database accounts to the IDM
+	 * @param options
+	 */
+	@Command({
+		command: 'migrate',
+		description: 'Add all database users to the IDM.',
+		options: [
+			...KeycloakConsole.retryFlags,
+			{
+				flags: '-q, --query',
+				description: 'Limit migration to accounts with username partially matching the query string.',
+				required: false,
+				defaultValue: undefined,
+			},
+			{
+				flags: '-s, --skip',
+				description: 'Skip the first "s" accounts during migration. Default is 0.',
+				required: false,
+				defaultValue: undefined,
+			},
+			{
+				flags: '-v, --verbose',
+				description: 'Output new id during the migration. Default false',
+				required: false,
+				defaultValue: false,
+			},
+		],
+	})
+	async migrate(options: IRetryOptions & IMigrationOptions): Promise<void> {
+		await this.repeatCommand(
+			'migrate',
+			async () => {
+				let position = options.skip ?? 0;
+				let count = 1;
+				this.console.start();
+				let success = 0;
+				while (count > 0) {
+					// eslint-disable-next-line no-await-in-loop
+					const res = await this.keycloakConfigurationUc.migrate(position, options.query);
+					count = res.amount;
+					position += res.amount;
+					res.errors.map((m) => this.console.warn(m));
```

Comment:
Map will return an array of size `errors.length`, maybe this will lead to memory problems? ForEach can be used instead.